server/api.js

```diff
@@ -4,15 +4,140 @@ import pool from "./db";
 
 const router = Router();
 
+// Get all the tasks
 router.get("/tasks", (req, res) => {
 	pool.query("SELECT * FROM tasks").then((result) => {
-		// console.log(result.rows);
 		return res.send(result.rows);
 	});
 });
-
 router.get("/", (_, res) => {
-	return res.json({ message: "Hello, from the api route!" });
+	res.json({ message: "Hello, from the api route!" });
 });
 
+router.post("/tasks", (req, res) => {
+	const { task_title, due_date, evidence, status_title, element_title } =
+		req.body;
+	let params = [];
+	if (!task_title) {
+		return res
+			.status(404)
+			.send({ success: false, message: "Task title cannot be empty" });
+	} else {
+		params.push(task_title);
+	}
+	if (due_date.length > 0) {
+		params.push(due_date);
+	} else {
+		params.push("");
+	}
+	if (evidence.length > 0) {
+		params.push(evidence);
+	} else {
+		params.push("");
+	}
+	if (status_title.length > 0) {
+		pool
+			.query("SELECT Status_id FROM status WHERE Status_title = $1", [
+				status_title,
+			])
+			.then((result) => {
+				params.push(result.rows[0].status_id);
+			})
+			.catch((err) => console.log(err));
+	} else {
+		res
+			.status(404)
+			.send({ success: false, message: "status_title cannot be empty" });
+	}
+	if (element_title.length > 0) {
+		pool
+			.query("SELECT Element_id FROM elements WHERE Element_title = $1", [
+				element_title,
+			])
+			.then((result) => {
+				params.push(result.rows[0].element_id);
+			})
+			.catch((err) => console.log(err));
+	} else {
+		res
+			.status(404)
+			.send({ success: false, message: "element_title cannot be empty" });
+	}
+
+	pool
+		.query(
+			"INSERT INTO tasks (Task_title,Due_date, Evidence, Status_id, Element_id) VALUES ($1,$2,$3,$4,$5)",
+			params
+		)
+		.then(() => {
+			pool.query("SELECT * FORM tasks").then((result) => res.send(result.rows));
+		});
+});
 export default router;
+
+//Delete a task
+
+router.delete("/:task_title", (req, res) => {
+	const title = req.params.task_title;
+	pool
+		.query("DELETE  FROM tasks WHERE Task_title = $1", [title])
```

Comment:
ok so now we are still not using user information to fetch their particular tasks, but I can imagine there can be several tasks in the table "tasks" with the same title, isn't task id a better approach to delete a task?

Reply:
Are we adding the ${id} to the fetch from the front end as params ?